cpp/kiss_icp/3rdparty/find_dependencies.cmake

```diff
@@ -20,41 +20,24 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
 # Silence timestamp warning
 if(CMAKE_VERSION VERSION_GREATER 3.24)
   cmake_policy(SET CMP0135 OLD)
 endif()
 
-if(USE_SYSTEM_EIGEN3)
-  find_package(Eigen3 QUIET NO_MODULE)
-endif()
-if(NOT USE_SYSTEM_EIGEN3 OR NOT TARGET Eigen3::Eigen)
-  set(USE_SYSTEM_EIGEN3 OFF)
-  include(${CMAKE_CURRENT_LIST_DIR}/eigen/eigen.cmake)
-endif()
-
-if(USE_SYSTEM_SOPHUS)
-  find_package(Sophus QUIET NO_MODULE)
-endif()
-if(NOT USE_SYSTEM_SOPHUS OR NOT TARGET Sophus::Sophus)
-  set(USE_SYSTEM_SOPHUS OFF)
-  include(${CMAKE_CURRENT_LIST_DIR}/sophus/sophus.cmake)
-endif()
+function(find_external_dependecy PACKAGE_NAME TARGET_NAME INCLUDED_CMAKE_PATH)
+  string(TOUPPER ${PACKAGE_NAME} PACKAGE_NAME_UP)
+  set(USE_FROM_SYSTEM_OPTION "USE_SYSTEM_${PACKAGE_NAME_UP}")
+  if(${${USE_FROM_SYSTEM_OPTION}})
+    find_package(${PACKAGE_NAME} QUIET NO_MODULE)
+  endif()
+  if(NOT ${${USE_FROM_SYSTEM_OPTION}} OR NOT TARGET ${TARGET_NAME})
+    set(${${USE_FROM_SYSTEM_OPTION}} OFF PARENT_SCOPE)
```

Comment:
Here the set function will not do the intended task. Basically, we want to set the value of the variable, which called USE_SYSTEM_${PACKAGE_NAME_UP}, to OFF in the parent scope. When you use two dollar signs, you will get the value hold by variable in the parent scope. However only if you use one dollar sign then you can actually obtain the name of the variable(USE_SYSTEM_${PACKAGE_NAME_UP}) and set it in 35 to OFF. So it should be as below

`set(${USE_FROM_SYSTEM_OPTION} OFF PARENT_SCOPE)`

Reply:
Good catch! I was blindly replacing it.